src/pages/PoolDetails/PoolDetailsStats.tsx

```diff
@@ -0,0 +1,229 @@
+import { Trans } from '@lingui/macro'
+import Column from 'components/Column'
+import CurrencyLogo from 'components/Logo/CurrencyLogo'
+import Row from 'components/Row'
+import { DeltaArrow, formatDelta } from 'components/Tokens/TokenDetails/Delta'
+import { PoolData } from 'graphql/thegraph/PoolData'
+import { useCurrency } from 'hooks/Tokens'
+import { useColor } from 'hooks/useColor'
+import { useScreenSize } from 'hooks/useScreenSize'
+import { ReactNode, useMemo } from 'react'
+import { Text } from 'rebass'
+import styled, { css } from 'styled-components'
+import { BREAKPOINTS } from 'theme'
+import { ThemedText } from 'theme/components'
+import { NumberType, useFormatter } from 'utils/formatNumbers'
+
+const HeaderText = styled(Text)`
+  font-weight: 485;
+  font-size: 24px;
+  line-height: 36px;
+  @media (max-width: ${BREAKPOINTS.lg - 1}px) {
+    width: 100%;
+  }
+`
+
+const StatsWrapper = styled(Column)`
+  gap: 24px;
+  margin: 0 48px 0 auto;
+  padding: 20px;
+  border-radius: 20px;
+  background: ${({ theme }) => theme.surface2};
+  width: 22vw;
+  min-width: 360px;
+
+  @media (max-width: ${BREAKPOINTS.lg - 1}px) {
+    margin: 44px 0px;
+    width: 100%;
+    flex-direction: row;
+    background: ${({ theme }) => theme.surface1};
+    flex-wrap: wrap;
+    padding: 20px 0px;
+    justify-content: space-between;
+  }
+`
+
+const StatItemColumn = styled(Column)`
+  gap: 8px;
+  flex: 1;
+  min-width: 180px;
+`
+
+const PoolBalanceSymbols = styled(Row)`
+  justify-content: space-between;
+
+  @media (max-width: ${BREAKPOINTS.lg - 1}px) {
+    flex-direction: column;
+  }
+`
+
+const PoolBalanceTokenNames = styled(Row)`
+  font-weight: 485;
+  font-size: 18px;
+  line-height: 24px;
+  width: max-content;
+
+  @media (max-width: ${BREAKPOINTS.lg - 1}px) {
+    font-size: 20px;
+    line-height: 28px;
+    width: 100%;
+  }
+`
+
+const leftBarChartStyles = css`
+  border-top-left-radius: 5px;
+  border-bottom-left-radius: 5px;
+  border-right: 1px solid ${({ theme }) => theme.surface2};
+`
+
+const rightBarChartStyles = css`
+  border-top-right-radius: 5px;
+  border-bottom-right-radius: 5px;
+  border-left: 1px solid ${({ theme }) => theme.surface2};
+`
+
+const BalanceChartSide = styled.div<{ percent: number; $color: string; isLeft: boolean }>`
+  height: 8px;
+  width: ${({ percent }) => percent * 100}%;
+  background: ${({ $color }) => $color};
+  ${({ isLeft }) => (isLeft ? leftBarChartStyles : rightBarChartStyles)}
+`
+
+interface PoolDetailsStatsProps {
+  poolData: PoolData
+  isReversed: boolean
+  chainId?: number
+}
+
+export function PoolDetailsStats({ poolData, isReversed, chainId }: PoolDetailsStatsProps) {
+  const isScreenSize = useScreenSize()
+  const screenIsNotLarge = isScreenSize['lg']
+  const { formatNumber } = useFormatter()
+
+  // We can't wrap this in a useMemo hook because useCurrency is also a hook
```

Comment:
is there a reason you can't do this?

```
const currencyA = useCurrency(poolData?.token0?.id, chainId) ?? undefined
const currencyB = useCurrency(poolData?.token1?.id, chainId) ?? undefined
const currencies = useMemo(() => ([currencyA, currencyB]), [currencyA, currencyB])
```

Reply:
but honestly i don't think you even need to wrap in an array at all here given you access by index later anyway?